src/lib/es2015.iterable.d.ts

```diff
@@ -188,26 +189,6 @@ interface WeakSetConstructor {
     new <T extends object>(iterable: Iterable<T>): WeakSet<T>;
 }
 
-interface Promise<T> { }
-
-interface PromiseConstructor {
-    /**
-     * Creates a Promise that is resolved with an array of results when all of the provided Promises
-     * resolve, or rejected when any Promise is rejected.
-     * @param values An array of Promises.
-     * @returns A new Promise.
-     */
-    all<TAll>(values: Iterable<TAll | PromiseLike<TAll>>): Promise<TAll[]>;

```

Comment:
By removing this, you can no longer take an `Iterable` for `Promise.all` or `Promise.race`. Consider:

```ts
function* g() {
  yield Promise.resolve(1);
  yield Promise.resolve(2);
}
Promise.all(g()).then(results => {
});
```
We either need to structure the files in such a way as to prevent this, or customize the ordering of lib files when we bind to enforce the correct order.